packages/local_auth/local_auth_platform_interface/lib/local_auth_platform_interface.dart

```diff
@@ -0,0 +1,116 @@
+
+  /// The default instance of [LocalAuthPlatform] to use.
+  ///
+  /// Defaults to [DefaultLocalAuthPlatform].
+  static LocalAuthPlatform get instance => _instance;
+
+  /// Platform-specific plugins should set this with their own platform-specific
+  /// class that extends [LocalAuthPlatform] when they register themselves.
+  static set instance(LocalAuthPlatform instance) {
+    PlatformInterface.verifyToken(instance, _token);
+    _instance = instance;
+  }
+
+  /// Authenticates the user with biometrics available on the device while also
+  /// allowing the user to use device authentication - pin, pattern, passcode.
+  ///
+  /// Returns a [Future] holding true, if the user successfully authenticated,
+  /// false otherwise.
+  ///
+  /// [localizedReason] is the message to show to user while prompting them
+  /// for authentication. This is typically along the lines of: 'Please scan
+  /// your finger to access MyApp.'. This must not be empty.
+  ///
+  /// [useErrorDialogs] = true means the system will attempt to handle user
+  /// fixable issues encountered while authenticating. For instance, if
+  /// fingerprint reader exists on the phone but there's no fingerprint
+  /// registered, the plugin will attempt to take the user to settings to add
+  /// one. Anything that is not user fixable, such as no biometric sensor on
+  /// device, will be returned as a [PlatformException].
+  ///
+  /// [stickyAuth] is used when the application goes into background for any
+  /// reason while the authentication is in progress. Due to security reasons,
+  /// the authentication has to be stopped at that time. If stickyAuth is set
+  /// to true, authentication resumes when the app is resumed. If it is set to
+  /// false (default), then as soon as app is paused a failure message is sent
+  /// back to Dart and it is up to the client app to restart authentication or
+  /// do something else.
+  ///
+  /// Provide [authStrings] if you want to
+  /// customize messages in the dialogs.
+  ///
+  /// Setting [sensitiveTransaction] to true enables platform specific
+  /// precautions. For instance, on face unlock, Android opens a confirmation
+  /// dialog after the face is recognized to make sure the user meant to unlock
+  /// their phone.
+  ///
+  /// Setting [biometricOnly] to true prevents authenticates from using non-biometric
+  /// local authentication such as pin, passcode, and passcode.
+  ///
+  /// Throws an [PlatformException] if there were technical problems with local
+  /// authentication (e.g. lack of relevant hardware). This might throw
+  /// [PlatformException] with error code [otherOperatingSystem] on the iOS
+  /// simulator.
+  Future<bool> authenticate({
+    required String localizedReason,
+    bool useErrorDialogs = true,
+    bool stickyAuth = false,
+    required Map<String, String> authStrings,
+    bool sensitiveTransaction = true,
+    bool biometricOnly = false,
```

Comment:
Let's future-proof this based on lessons learned in other federated plugins by moving all of these booleans into a class called something like `AuthenticationOptions`. That way we can add new parameters without the breaking-change-or-duplicate method dance.

Reply:
Agreed, that would be much better. Changed.